lib/updateAccessorComponentTypes.js

```diff
@@ -0,0 +1,67 @@
+'use strict';
+var Cesium = require('cesium');
+var addToArray = require('./addToArray');
+var ForEach = require('./ForEach');
+var readAccessorPacked = require('./readAccessorPacked');
+
+var ComponentDatatype = Cesium.ComponentDatatype;
+var WebGLConstants = Cesium.WebGLConstants;
+
+module.exports = updateAccessorComponentTypes;
+
+/**
+ * Update accessors referenced by JOINTS_0 and WEIGHTS_0 attributes to use correct component types.
+ *
+ * @param {Object} gltf A javascript object containing a glTF asset.
+ * @returns {Object} The glTF asset with compressed meshes.
+ *
+ * @private
+ */
+function updateAccessorComponentTypes(gltf) {
+    var componentType;
+    ForEach.accessorWithSemantic(gltf, 'JOINTS_0', function(accessorId) {
+        var accessor = gltf.accessors[accessorId];
+        componentType = accessor.componentType;
+        if (componentType === WebGLConstants.BYTE) {
+            convertType(gltf, accessor, componentType, ComponentDatatype.UNSIGNED_BYTE);
+        } else if (componentType !== WebGLConstants.UNSIGNED_BYTE
+                && componentType !== WebGLConstants.UNSIGNED_SHORT) {
+            convertType(gltf, accessor, componentType, ComponentDatatype.UNSIGNED_SHORT);
+        }
+    });
+    ForEach.accessorWithSemantic(gltf, 'WEIGHTS_0', function(accessorId) {
+        var accessor = gltf.accessors[accessorId];
+        componentType = accessor.componentType;
+        if (componentType === WebGLConstants.BYTE) {
+            convertType(gltf, accessor, componentType, ComponentDatatype.UNSIGNED_BYTE);
+        } else if (componentType === WebGLConstants.SHORT) {
+            convertType(gltf, accessor, componentType, ComponentDatatype.UNSIGNED_SHORT);
+        }
+    });
+
+    return gltf;
+}
+
+function convertType(gltf, accessor, componentType, updatedComponentType) {
+    var data = ComponentDatatype.createTypedArray(componentType, readAccessorPacked(gltf, accessor));
+    var typedArray = ComponentDatatype.createTypedArray(updatedComponentType, data);
+    var newBuffer = Buffer.from(typedArray.buffer);
```

Comment:
`Buffer.from` will not work from when this file is bundled in Cesium. I wonder if doing `Uint8Array(typedArray.buffer)` will work instead and across different versions of Node.

Reply:
This worked in Node 8 and Cesium, but failed in Node 6. A quick fix would be to convert `Uint8Array` buffer view data into a `Buffer` in `mergeBuffers`.

Reply:
I ran this in Node 6.10 and it works. Likewise it works in CI with 6.14.

Reply:
Using 6.14.3 I get:

> C:\Code\gltf-pipeline>node bin/gltf-pipeline.js -i test/RiggedSimple.gltf
> TypeError: "list" argument must be an Array of Buffers

Reply:
Upgraded and was able to duplicate. Should be fixed now.